modules/create_timepoint/jsx/createTimepointIndex.js

```diff
@@ -0,0 +1,399 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Panel from 'Panel';
+import Loader from 'Loader';
+import swal from 'sweetalert2';
+
+/**
+ * Create Timepoint.
+ *
+ * @description form for create timepoint.
+ *
+ * @author Alizée Wickenheiser
+ * @version 1.0.0
+ *
+ */
+class CreateTimepoint extends React.Component {
+  /**
+   * @constructor
+   * @param {object} props - React Component properties
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {
+        pscid: '',
+        dccid: '',
+        visit: '',
+        languages: '',
+      },
+      form: {
+        display: {
+          subproject: false,
+          visit: false,
+          psc: false,
+          project: false,
+          languages: false,
+        },
+        options: {
+          subproject: {},
+          visit: {},
+          psc: {},
+          project: {},
+          languages: {},
+          required: {
+            subproject: true,
+            psc: true,
+            project: true,
+            visit: true,
+            languages: true,
+          },
+        },
+        value: {
+          subproject: null,
+          visit: null,
+          psc: null,
+          project: null,
+          languages: null,
+        },
+      },
+      storage: {
+        visit: {},
+      },
+      messages: [],
+      url: {
+        params: {
+          candID: '',
+          identifier: '',
+        },
+      },
+      success: false,
+    };
+
+    // Bind component instance to custom methods
+    this.fetchInitializerData = this.fetchInitializerData.bind(this);
+    this.handleVisitLabel = this.handleVisitLabel.bind(this);
+    this.collectParams = this.collectParams.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setForm = this.setForm.bind(this);
+  }
+
+  /**
+   * Executes after component mounts.
+   */
+  componentDidMount() {
+    this.collectParams().then(() => {
+      this.fetchInitializerData().then(() => {
+        this.setState({isLoaded: true});
+      });
+    });
+  }
+
+  /**
+   * Retrieve params from the browser URL and save it in state.
+   */
+  async collectParams() {
+    const url = new URL(window.location.href);
+    const state = JSON.parse(JSON.stringify(this.state));
+    state.url.params = {
+      candID: url.searchParams.get('candID'),
+      identifier: url.searchParams.get('identifier'),
+    };
+    state.data.dccid = state.url.params.candID;
+    await this.setState(state);
```

Comment:
I don't think `this.setState` returns a promise, so I'm not sure this works like this.

Reply:
I think you would either want to do:
`return new Promise((resolve) => this.setState(state, resolve()))`

or if you want to use async/await it would be something like described here:
https://medium.com/front-end-weekly/async-await-with-react-lifecycle-methods-802e7760d802

Reply:
@HenriRabalais thanks. Please see latest commit.